### PR TITLE
bpo-46968: Check for 'sys/auxv.h' in the configure script

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue46968

### CODE REVIEW

File: Modules/faulthandler.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "Python.h"
+#include "Python.h" && defined(HAVE_SYS_AUXV_H)
```
Comment (pablogsal) [inline Modules/faulthandler.c]:
```suggestion
#include "Python.h"
```

**Review by oleksandr-pavlyk — APPROVED** [review]

**Comment by oleksandr-pavlyk** [comment]
Sorry about the issue. I was thinking of adding the test but then assumed existence of `linux/auxvec.h` implies existence of `sys/auxv.h` based on inspection of recent kernels.

File: Modules/faulthandler.c [hunk 2]
```diff
@@ -26,7 +26,7 @@
 #  define FAULTHANDLER_USE_ALT_STACK
 #endif
 
-#if defined(FAULTHANDLER_USE_ALT_STACK) && defined(HAVE_LINUX_AUXVEC_H)
+#if defined(FAULTHANDLER_USE_ALT_STACK) && defined(HAVE_LINUX_AUXVEC_H) && defined(HAVE_SYS_AUXV_H)
 #  include <linux/auxvec.h>
 #  include <sys/auxv.h>
```
Comment (vstinner) [inline Modules/faulthandler.c]:
```suggestion
#  include <linux/auxvec.h>       // AT_MINSIGSTKSZ
#  include <sys/auxv.h>           // getauxval()
```

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by vstinner** [comment]
> Sorry about the issue. I was thinking of adding the test but then assumed existence of linux/auxvec.h implies existence of sys/auxv.h based on inspection of recent kernels.

Same here. But the Linux ecosystem is always more complicated than what I expect :-)

**Pull request merged**